test/runtests.jl

```diff
@@ -424,6 +427,10 @@ cd(@__DIR__) do
         Test.record(o_ts, fake)
         Test.pop_testset()
     end
+
+    Base.get_bool_env("CI", false) &&
+        open(io -> write_testset_json(io, o_ts), "results.json", "w")
```

Comment:
Probably good to give an indication it's been written
```suggestion
    if Base.get_bool_env("CI", false)
        open(io -> write_testset_json(io, o_ts), "results.json", "w")
        @info "CI detected. Test results written to `results.json` ($(Base.format_bytes(filesize("results.json"))))" 
    end
```

Reply:
Given CI is green this could be for a future PR

Reply:
Maybe we should add that with some information to where we are writing the file since our upload step is still failing :)